src/test/java/com/google/jenkins/plugins/computeengine/ComputeEngineCloudWindowsIT.java

```diff
@@ -0,0 +1,299 @@
+
+        // There should be a planned node
+        assertEquals(logs(), 1, planned.size());
+
+        // Wait for the node creation to finish
+        String name = planned.iterator().next().future.get().getNodeName();
+
+        // There should be no warning logs
+        assertEquals(logs(), false, logs().contains("WARNING"));
+
+        Instance i = cloud.client.getInstance(projectId, ZONE, name);
+
+        // The created instance should have 3 labels
+        assertEquals(logs(),3, i.getLabels().size());
+
+        // Instance should have a label with key CONFIG_LABEL_KEY and value equal to the config's name prefix
+        assertEquals(logs(), validInstanceConfiguration1().namePrefix, i.getLabels().get(ComputeEngineCloud.CONFIG_LABEL_KEY));
+        // proper id label to properly count instances
+        assertEquals(logs(), String.valueOf(cloud.name.hashCode()), i.getLabels().get(ComputeEngineCloud.CLOUD_ID_LABEL_KEY));
+    }
+
+    /**
+     *
+     * @param publicKey Public Key to be put into the authorized_key file in our windows VM for SSH authentication
+     * @return startupScript String startup script to run
+     */
+    private static String createStartupScript(String publicKey) {
+        String startupScript = "Stop-Service sshd\n" +
+                "$username = " + "\"" + WINDOWS_USER + "\"" + "\n" +
+                "$ConfiguredPublicKey = "  + "\"" + publicKey  + "\"" + "\n" +
+                "Write-Output \"Second phase\"\n" +
+                "# We are in the second phase of startup where we need to set up authorized_keys for the specified user.\n" +
+                "$UserSid = Get-WmiObject win32_useraccount -Filter \"name = '$username'\" | select-object sid -ExpandProperty SID\n" +
+                "$UserProfilePath = Get-ItemProperty -Path  \"HKLM:\\Software\\Microsoft\\Windows NT\\CurrentVersion\\ProfileList\\$UserSid\" -Name ProfileImagePath | select-object profileimagepath -ExpandProperty ProfileImagePath\n" +
+                "# Create the .ssh folder and authorized_keys file.\n" +
+                "mkdir $UserProfilePath\\.ssh\n" +
+                "Set-Content -Path $UserProfilePath\\.ssh\\authorized_keys -Value $ConfiguredPublicKey\n" +
+                "# Fix up permissions on authorized_keys\n" +
+                "Import-Module \"$env:PROGRAMFILES\\OpenSSH-Win64\\OpenSSHUtils.psd1\" -Force\n" +
+                "Repair-AuthorizedKeyPermission -FilePath  $UserProfilePath\\.ssh\\authorized_keys\n" +
+                "Restart-Service sshd";
+
+        return startupScript;
+    }
+
+    private static InstanceConfiguration validInstanceConfiguration1() throws Exception{
+        GoogleKeyPair kp = GoogleKeyPair.generate("");
+        // Have to reformat since GoogleKeyPair's format is for metadata server and not typical public key format
+        String publicKey = kp.getPublicKey().trim().substring(1);
+
+        String startupScript = createStartupScript(publicKey);
```

Comment:
createStartupScript is only called here, suggest using String.format instead and removing createStartupScript.

Reply:
done